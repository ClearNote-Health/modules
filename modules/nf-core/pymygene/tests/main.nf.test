nextflow_process {

    name "Test Process PYMYGENE"
    script "../main.nf"
    process "PYMYGENE"

    tag "modules"
    tag "modules_nfcore"
    tag "pymygene"

    test("pymygene - txt") {
        when {
            process {
                """
                // Create a temporary file with the first column of the input file, without header
                source = params.test_data['mus_musculus']['genome']['rnaseq_matrix']
                def tmp = new File('tmp.tsv')
                tmp.withWriter { writer ->
                    new URL(source).withReader { reader ->
                        reader.readLine() // Skip the header line
                        reader.eachLine { line, lineNumber ->
                            def cols = line.split('\\t')
                            writer.write(cols[0] + '\\n')
                        }
                    }
                }
                input[0] = [
                    [id : 'test'],
                    file('tmp.tsv')
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.tsv).match("tsv") },
                { assert snapshot(process.out.gmt).match("gmt") },
                { assert snapshot(process.out.versions).match("versions") }
            )
        }
    }
}
