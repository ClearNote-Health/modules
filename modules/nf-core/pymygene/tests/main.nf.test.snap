{
    "versions": {
        "content": [
            [
                "versions.yml:md5,1f594a9a1d1fc1fb1803a116de432465"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-19T16:30:19.351166"
    },
    "tsv": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.tsv:md5,56b545110721589ea7e0ad5443f68743"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-19T16:30:19.325275"
    },
    "gmt": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.gmt:md5,3084e411ed66e2d0f809e8f0b41213a5"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-19T16:30:19.341297"
    }
}